main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -173,8 +179,32 @@ DataTableView.prototype._renderPagingControls = function(pageSizeControls, pagin
     previousPage.addClass("inaction");
   }
 
-  $('<span>').addClass("viewpanel-pagingcount").html(" " + from + " - " + to + " ").appendTo(pagingControls);
+  var pageControlsSpan = $('<span>').attr("id", "viewpanel-paging-current");
+  
+  var pageInputSize = 20 + (8 * ui.dataTableView._lastPageNumber.toString().length);
+  var currentPageInput = $('<input type="number">')
+    .change(function(evt) { self._onChangeGotoPage(this, evt); })
+    .keydown(function(evt) { self._onKeyDownGotoPage(this, evt); })
+    .attr("id", "viewpanel-paging-current-input")
+    .attr("min", 1)
+    .attr("max", self._lastPageNumber)
+    .attr("required", "required")
+    .val(self._currentPageNumber)
+    .css("width", pageInputSize +"px");
+    
+  pageControlsSpan.append($.i18n('core-views/goto-page', '<span id="currentPageInput" />', self._lastPageNumber));
+  pageControlsSpan.appendTo(pagingControls);
 
+  $('span#currentPageInput').replaceWith($(currentPageInput));
+  
+  if(self._refocusPageInput == true) { 
+    self._refocusPageInput = false;
+    var currentPageInput = $('input#viewpanel-paging-current-input');
+    currentPageInput.ready(function(evt) { 
+      (new Promise(resolve => setTimeout(resolve, 1000))).then(currentPageInput.focus()); 
```

Comment:
There is still a `setTimeout` here :)

Reply:
> There is still a `setTimeout` here :)

@wetneb: yes, but this one is very clean. It's to make the « scrolling » thru pages not too fast. If you have a better delay strategy, please advise me.

Regards,
   Antoine

Reply:
ok fair enough, I will test this as a user now, and might go back to this later on when I understand its role better.

Reply:
> ok fair enough, I will test this as a user now, and might go back to this later on when I understand its role better.

I'm all available to explain everything to speed up this. I'm very confident that all the requirements are there. We can also go thru the other features that are still waiting for approval.

Regards,
   Antoine